api/v1beta1/common_types.go

```diff
@@ -164,6 +164,14 @@ type ServiceOptions struct {
 	// Labels to be added for the Service.
 	// +optional
 	Labels map[string]string `json:"labels,omitempty"`
+
+	// Supports "ClientIP" and "None". Used to maintain session affinity. Enable client IP based session affinity. Must be ClientIP or None. Defaults to None
+	// +optional
+	SessionAffinity string `json:"sessionAffinity,omitempty"`
+
+	// sessionAffinityConfig contains the configurations of session affinity.
+	// +optional
+	SessionAffinityConfig corev1.SessionAffinityConfig `json:"sessionAffinityConfig,omitempty"`
```

Comment:
```suggestion
	SessionAffinityConfig *corev1.SessionAffinityConfig `json:"sessionAffinityConfig,omitempty"`
```

Making this field a pointer, makes it easier for us to know whether the user provided one or not!

Reply:
Makes sense